From bf14d07061b46604781fe23a9a2f62b7853e4407 Mon Sep 17 00:00:00 2001
From: Zhen Chen <chenzhen@rock-chips.com>
Date: Thu, 10 Jan 2019 11:02:03 +0800
Subject: [PATCH] arm64: dts: rockchip: add 'gpu_power_model' feature with DDK
 r14 for linux

Since the mali.ko is *not* upgraded by userspace, rk3399 linux hope keep
mali driver version is the same with userspace.

Change-Id: Id5af615f1f7b0c95587bc5faeeeff35251797e1a
Signed-off-by: Zhen Chen <chenzhen@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi
index ea4470e08a86..a153bc06facc 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi
@@ -152,6 +152,15 @@
 	};
 };
 
+/* HACK: keep MALI version on linux */
+&gpu_power_model {
+	// for DDK r14.
+	voltage = <900>;
+	frequency = <500>;
+	static-power = <300>;
+	dynamic-power = <396>;
+};
+
 &pvtm {
 	status = "okay";
 };
-- 
2.35.3

